src/dsp_tools/xmllib/models/root.py

```diff
@@ -51,11 +54,39 @@ def add_resource_optional(self, resource: Resource | None) -> XMLRoot:
 
     def serialise(self) -> etree._Element:
         root = self.make_root()
-        root = append_permissions(root)
+        permissions = XMLPermissions().get()
+        root.extend(permissions)
         serialised_resources = [x.serialise() for x in self.resources]
         root.extend(serialised_resources)
         return root
 
     def write_file(self, filepath: str | Path) -> None:
+        """
+        Write the finished XML to a file.
+
+        Args:
+            filepath: where to save the file
+
+        Warning:
+            if the XML is not valid, according to the schema
+        """
         root = self.serialise()
-        write_xml(root, filepath)
+        etree.indent(root, space="    ")
+        xml_string = etree.tostring(
+            root,
+            encoding="unicode",
+            pretty_print=True,
+            doctype='<?xml version="1.0" encoding="UTF-8"?>',
+        )
+        xml_string = xml_string.replace(r"\'", "'")
+        with open(filepath, "w", encoding="utf-8") as f:
+            f.write(xml_string)
+        try:
+            validate_xml_file(input_file=filepath)
+            print(f"The XML file was successfully saved to {filepath}")
+        except BaseError as err:
+            msg = (
+                f"The XML file was successfully saved to {filepath}, "
+                f"but the following Schema validation error(s) occurred: {err.message}"
+            )
+            warnings.warn(DspToolsUserWarning(msg))
```

Comment:
I copy pasted this part of the code, Johannes implemented it because it solved a problem. He does however not remember why. I am hesitant to just try this out because the code is not tested well and this has worked...
I would vote that we leave it as is because it works and we carefully examine this separately and test it well.